code/Switch.tsx

```diff
@@ -279,17 +280,21 @@ function _Switch(props) {
                 data-switch-id={sanitizedIdentifier}
             >
                 <Frame background={null} width="100%" height="100%">
-                    <AutoAnimatedState
-                        source={children[previousIndexRef.current]}
-                        target={children[currentIndexRef.current]}
-                        transitionPropsForElement={transitionPropsForElement}
-                        direction={direction}
-                        sourceParentSize={size}
-                        targetParentSize={size}
-                        morphCodeComponentPropsOnly={
-                            morphCodeComponentPropsOnly
-                        }
-                    />
+                    <PreventLayoutIdGeneration>
```

Comment:
@tisho perhaps this is not necessary since you mentioned you're already touching `_canMagicMotion`?